core/model/modx/modfilehandler.class.php

```diff
@@ -514,6 +514,70 @@ public function create($mode = '') {
     }
 
     /**
+     * Iterates over a modDirectory object and returns an array of all containing files,
+     * optionally runs recursive, filters by file extension(s) and sorts the resulting list according to a specified sort flag
+     *
+     * @param array $options Options for iterating the directory.
+     *      boolean recursive If also subfolders should be scanned for files
+     *      boolean ignorehidden If folders and files starting with a dot . (hidden files/folders in unix envirionments) should be ignored, defaults to true
+     *      boolean sort If the resulting filelist array should be sorted with the specified flag like SORT_ASC, SORT_DESC, SORT_REGULAR, SORT_NATURAL, SORT_NUMERIC
+     *      string extensions Comma separated list of file extensions to filter files by
+     *      
+     * @return array of filepaths
+     */
+    public function getFiles($options = array()) {
+        $options = array_merge(array(
+            'recursive' => false,
+            'ignorehidden' => true,
+            'sort' => false,
+            'extensions' => ''
+        ), $options);
+
+        $files = array();
+        $extensions = explode(',', $options['extensions']);
+        
+        if ($options['recursive']) {
+            $iterator = new RecursiveDirectoryIterator($this->path);
+
+            foreach (new RecursiveIteratorIterator($iterator) as $file) {
+                $dirname = explode(DIRECTORY_SEPARATOR, $file->getPath()); // PHP Strict warning if we put end() around here
+                
+                if (($options['ignorehidden'] ? (substr($file->getFilename(), 0, 1) !== '.' && substr(end($dirname), 0, 1) !== '.') : true) && $file->isFile()) {
```

Comment:
I don't understand what is sooo confusing with those lines? He is just using the Iterator class.